### PR TITLE
Clean up Keycloak collaborators/resources

### PR DESCRIPTION
**Pull request by alexeykazakov**
As soon as https://github.com/fabric8-services/fabric8-wit/pull/2129 is merged and deployed to prod we no longer need to manage Keycloak resources and permissions.

This PR removes everything related to Keycloak AuthZ.

- [x] Clean up KC AuthZ code
- [x] Clean up tests 
- [x] Drop KC space resources table

Part of https://github.com/fabric8-services/fabric8-auth/issues/359
fixes https://github.com/fabric8-services/fabric8-auth/issues/539

### CODE REVIEW

**Comment by alien-ike** [comment]
### Ike Plugins (test-keeper)

Thank you @alexeykazakov for this contribution!



It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


For more information please head over to official [documentation](http://arquillian.org/ike-prow-plugins/#_test_keeper_plugin). You can find there how to configure the plugin.

**Comment by fabric8cd** [comment]
@alexeykazakov snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-533-1`

**Comment by fabric8cd** [comment]
@alexeykazakov snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-533-3`

**Comment by alexeykazakov** [comment]
[test]

**Comment by alexeykazakov** [comment]
[test]

**Comment by sbose78** [comment]
[test]

**Comment by fabric8cd** [comment]
@alexeykazakov snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-533-4`

File: design/spaces.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ var spaceResourceData = a.Type("SpaceResourceData", func() {
 	a.Attribute("resourceID", d.String, "Keycloak Resource ID associated with this Space")
 	a.Attribute("permissionID", d.String, "Keycloak Permission ID associated with this Space")
 	a.Attribute("policyID", d.String, "Keycloak Policy ID associated with this Space")
-	a.Required("resourceID", "permissionID", "policyID")
+	a.Required("resourceID")
```
[the remark this answers is not in the document]
Comment (alexeykazakov) [reply design/spaces.go]: Yeah... it should be fine to remove them completely.

**Comment by sbose78** [comment]
https://ci.centos.org/job/devtools-fabric8-auth/1392/console

```
--- FAIL: TestUsersController (2.81s)
    --- FAIL: TestUsersController/TestListUsersNotModifiedUsingIfModifiedSinceHeader (0.02s)
    	users_testing.go:690: invalid response status code: got 200, expected 304
FAIL
```
@xcoulon :) 

[test]

**Pull request merged**